Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -38,9 +39,12 @@ public abstract class BaseJobRunner
             JobName = jobName;
             _analytics = analytics;
 
-            JobBinariesPath = Path.Combine(Environment.JobsBinariesPath, jobsTypePath, jobName);
-            JobTempPath = Path.Combine(Environment.TempPath, Constants.JobsPath, jobsTypePath, jobName);
-            JobDataPath = Path.Combine(Environment.DataPath, Constants.JobsPath, jobsTypePath, jobName);
+            JobBinariesPath = Path.Combine(basePath, jobName);
+
+            // add a 10 char long hash of the jobBinaries location to the log folder path
+            var hash = JobBinariesPath.GetMD5Hash(10);
```

Comment:
Why is this new hash necessary? It's breaking test `InPlaceJobRunsInPlace`, which checks that the path it runs from contains `App_Data\jobs\triggered\`. But note that there is already a random component to the temp path, e.g. `TEMP\jobs\triggered\joba\ege02v04.osz`.

More generally, I'm guessing you had added that when both jobs by the same name were allowed to exist, but that it may no longer be relevant now that you enforce uniqueness?

Reply:
your guess is correct. I forgot to undo this change. Just pushed a fix for it. also a change for the delete logic. Now it's

```cs
if (_settings.RunFromLocalZip() || _settings.RunFromRemoteZip())
{
     WriteJobManager.DeleteJob(jobName);
}
else if (PrimaryJobManager.HasJob(jobName))
{
    PrimaryJobManager.DeleteJob(jobName);
}
else
{
    SecondaryJobManager.DeleteJob(jobName);
}
```